src/ui/options/components/permissions.tsx

```diff
@@ -0,0 +1,37 @@
+import { Setter, Show, createSignal } from 'solid-js';
+import browser from 'webextension-polyfill';
+import styles from './components.module.scss';
+import { t } from '@/util/i18n';
+
+export default function Permissions() {
+	const [perms, setPerms] = createSignal(true);
+	hasPermissions(setPerms);
+
+	return (
+		<Show when={!perms()}>
+			<div class={styles.permissionsPopup}>
+				<button
+					type="button"
+					onClick={() => {
+						// This is inline and not async, as when it was not there were some issues with safari isTrusted
+						browser.permissions
+							.request({
```

Comment:
Would we extract the permissions into a variable for keeping things DRY?

Reply:
Sure can, give me a bit

Reply:
Done